src/stores/WalletHistoryStore.js

```diff
@@ -0,0 +1,63 @@
+import { observable, action } from 'mobx';
+import _ from 'lodash';
+
+import Transaction from './models/Transaction';
+import { queryAllTransactions } from '../network/graphQuery';
+import { SortBy, TransactionType } from '../constants';
+
+
+export default class WalletHistoryStore {
+  @observable fullList = []
+  @observable list = []
+  @observable orderBy = 'createdTime'
+  @observable direction = SortBy.Descending
+  @observable limit = 500
+  @observable skip = 0
+  @observable perPage = 10
+  @observable page = 0
+  @observable expanded = []
+
+  @action.bound
+  queryTransactions = async (orderBy = this.orderBy, direction = this.direction, limit = this.limit, skip = this.skip) => {
+    try {
+      const filters = [{ type: TransactionType.Transfer }];
+      const orderByObj = { field: orderBy, direction };
+
+      const result = await queryAllTransactions(filters, orderByObj, limit, skip);
+      this.fullList = _.map(result, (tx) => new Transaction(tx));
+      this.list = _.slice(this.fullList, 0, this.perPage);
+    } catch (error) {
+      console.error(error); // eslint-disable-line
+      this.fullList = [];
+      this.list = [];
+    }
+  }
+
+  @action.bound
+  onPageChange = (page) => {
```

Comment:
Ok, so just a thought here on how I would write this. I kinda don't like having to manage `state` of items in an array like this. I would do something like

`/WalletHistory/HistoryItem/index.js` or something along those lines
and do something like
```js
@observer
class HistoryItem extends Component {
  @observable open = false // OR state = { expanded: false }
  toggle = () => this.open = !this.open
  render() {
    return (
      <Fragment>
        <Item onClick={this.toggle}>
        </Item>
        {this.open && (
          <Drawer></Drawer>
        )}
      </Fragment>
    )
  }
}
```

Reply:
Then we can remove all the `this.expanded = []` from all these methods

Reply:
done, makes sense. I am keeping the expanded in the component state. is that ok?